app/configurator/configurator-state/reducer.tsx

```diff
@@ -1094,13 +1102,68 @@ export const reducer_: Reducer<ConfiguratorState, ConfiguratorStateAction> = (
 };
 
 /** Turn this on for the reducer to log state, action and result */
-const reducerLogging = false;
+const reducerLogging = true;
 const withLogging = <TState, TAction extends { type: unknown }>(
   reducer: Reducer<TState, TAction>
 ) => {
   return (state: Draft<TState>, action: TAction) => {
     const res = reducer(state, action);
+    console.log(`Action: ${action.type}`, action, res);
     return res;
   };
 };
 export const reducer = reducerLogging ? withLogging(reducer_) : reducer_;
+
+export function ensureDashboardLayoutAreCorrect(
+  draft: WritableDraft<ConfiguratorState>
+) {
+  if (
+    hasChartConfigs(draft) &&
+    draft.layout.type === "dashboard" &&
+    draft.layout.layout === "canvas"
+  ) {
+    const chartConfigKeys = draft.chartConfigs.map((c) => c.key).sort();
+
+    const canvasLayouts = draft.layout.layouts["lg"];
```

Comment:
Is this property always there? What if the size of the canvas was initialized as "md"?

Reply:
Right now this property is always there yes. The chart is always initialized with md.

Reply:
The responsive behavior could be checked later, but I think right now it works well, I've added https://github.com/visualize-admin/visualization-tool/issues/1567

Reply:
Thanks 💯